FFM-7245: Updates Git Exp and edits to original content (#1068)
harness/developer-hub: docs/feature-flags/ff-using-flags/manage-featureflags-in-git-repos.md

The synchronization between the Harness Platform and the `flags.yaml` file works in both directions:

* When you update the Harness Platform, the changes are committed to Git. Changes are synced to the remote repository immediately.
* When you commit changes to Git, the Harness Platform is periodically updated. The changes will be synced up to 5 min after you commit in Git.

If you don’t see the changes you made in Git reflected on the Harness Platform after approxymately 5 min, refresh the page.

:::caution
 Syncing changes between a remote file and the Harness Platform can take up to 5 mins. During this window the changes are commited to the remote file but not yet pulled and synced by the Harness Platform. Any changes made to the Harness Platform within that window trigger remote file updates, which overwrite the content of the remote file.
:::

## Turn on syncing with Git

You can turn the synchronization between the Harness Platform and Git on or off. The Git Experience icon is displayed on many pages, and you can toggle it on or off from any page where it appears.

:::caution
 Turning sync on triggers an immediate attempt to sync Harness Platform content to the remote file.

>  All the changes made to the remote file while sync was disabled are overwritten, which results in losing content or configuration not yet synced to the Platform.

Reviewer: ```suggestion
 Any pre-sync changes in the remote file are overwritten. This can result in losing content or configurations that have not been synced to the Platform.
```

Author: Great catch! I reworded a bit: 
Turning sync on triggers an immediate attempt to sync Harness content to the remote file. Any changes you made to the remote file before syncing are overwritten. This can result in losing content or configurations that are not yet synced to Harness.

:::

To turn on synchronization between the Harness Platform and Git: 

1. Ensure that Git Experience has been [enabled](/docs/platform/Git-Experience/configure-git-experience-for-harness-entities) for your project.

1. Select **Feature Flags**.

  In the top bar navigation, the Git repository and the branch you connected are displayed.

  ![The Sync with Git toggle highlighted](./static/5-manage-featureflags-in-git-repos-04.png)

>   *Figure 2: Toggle to turn on syncing with Git*

Reviewer: I don't see a Figure 1 in the PR build of this topic:

https://hdh.pr.harness.io/pr-1068/docs/feature-flags/ff-using-flags/manage-featureflags-in-git-repos/

I know our style guide says "Add figure titles to screenshots" but you might be the first person to do this :)

Author: Haha... I really don't think it makes sense to have figure titles in step procedures! They make sense in conceptual topics where you might actually refer to a figure from somewhere else in the doc, but in procedures they clutter up the steps making them less scannable/confusing... and you wouldn't refer to them inside a procedure... you'd just xref the heading of the procedure. So thanks for making that comment, I'm going to nuke them inside procedures (and ask for forgiveness later :-D )

1. Select the branch, and then toggle **Sync with Git** on. 

>     If you turn on only this button, each time you make a change to a flag on Harness, you are prompted to confirm which branch you want to commit to, and to add a commit message. For example:

Reviewer: ```suggestion
    When **Sync with Git** is enabled, you are prompted to confirm each change to a flag on Harness, including the commit branch and message. For example:
```

Author: Actually had to say it like that because the behavior is different if you toggle one or both. I added an extra line+link to clarify a bit.

    ```mdx-code-block
    <img src={git_2} alt="An example commit message." height="500" width="400" />
    ```
    *Figure 3: Example commit message*

> 1. Optionally, you can select the checkbox **Always commit to this branch and do not prompt for commit message**. 

Reviewer: I typically don't name UI elements like "checkbox" etc. Per the style guide: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide#UI-elements

Author: Actually on second look, I can delete checkbox here too, thanks!
